kubernetes/website
# Update website/update-imported-docs/update-imported-docs script for Python 3, PyYAML 5.1.2

The current version of the script works with Python 2.7, PyYAML 3.13, Go 1.12.  

This PR grew out of my attempts to run the website/update-imported-docs/update-imported-docs script using Python 3.7, **Go 1.13**, and PyYAML 5.1 -- Slack [thread](https://kubernetes.slack.com/archives/C1J0BPD2M/p1570125720198000).Using Go 1.13 caused numerous GOPATH issues.

Neither the script nor [documentation](https://kubernetes.io/docs/contribute/generate-ref-docs/kubernetes-components/) specify what version of Python the script was tested with, nor the version of Golang, Git, or PyYAML; docs page says Golang v 1.9 or later… yet Kubernetes development lists specific versions of Golang for specific versions of Kubernetes (see [list](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#go)).  My issue was using  Go 1.13, which Kubernetes does not support.

@tengqm agreed that the script should be updated to work with Python 3. I took this opportunity to "pythonify" the code.

Updating to the latest version of PyYAML required an update to remove a deprecated function. Looking at the code, I decided it would be better to use the Python tempfile library to create the temp directory rather than `os` and `shutil` libraries.

**This script was tested on Ubuntu 18 with Python 3.7.4, go1.12.10 linux/amd64, and PyYAML 5.1.2 installed in a virtual environment.**

Current version in master for comparison: https://github.com/kubernetes/website/blob/master/update-imported-docs/update-imported-docs

- Add .py extension to update-imported-docs file
- Update shebang for Python 3
- Add script comments: software versions, what the script does, and how to run it
- Update to use argparse.ArgumentParser() for command line args
- Change work_dir from hard-coded dir to tmp dir created using the tempfile library; remove shutil.rmtree because it was throwing permission errors and also no longer needed after switching
to temp dir created with tempfile.
- Enhance error handling
- Add check for Go installation
- Enhance check for PyYAML installation
- Change deprecated PyYAML load function to full_load https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation
- Modify to comply with PEP8

Partial #15682 

**Also updated the software version in kubernetes-components.md #before-you-begin**

Signed-off-by: Aimee Ukasick <aimeeu.opensource@gmail.com>

**STEPS TO TEST THE SCRIPT**
Scenario 1: PyYAML not installed.
1. (Optional) Create a virtual environment with Python = 3.7.4
2. `cd` to website/update-imported-script and execute the script `$ ./update-imported-docs.py reference.yml`  - you should see a message that says `Please ensure PyYAML package is installed (https://pypi.org/project/PyYAML/). This can be done, for example, by executing the following command: pip install PyYAML`
